source-local/src/androidMain/kotlin/tachiyomi/source/local/LocalSource.kt

```diff
@@ -225,16 +230,31 @@ actual class LocalSource(
         }
     }
 
-    private fun setMangaDetailsFromComicInfoFile(stream: InputStream, manga: SManga) {
-        val comicInfo = AndroidXmlReader(stream, StandardCharsets.UTF_8.name()).use {
+    private fun parseComicInfoFile(stream: InputStream): ComicInfo {
+        return AndroidXmlReader(stream, StandardCharsets.UTF_8.name()).use {
             xml.decodeFromReader<ComicInfo>(it)
         }
+    }
 
-        manga.copyFromComicInfo(comicInfo)
+    private fun setMangaDetailsFromComicInfoFile(stream: InputStream, manga: SManga) {
+        manga.copyFromComicInfo(parseComicInfoFile(stream))
+    }
+
+    private fun setChapterDetailsFromComicInfoFile(stream: InputStream, chapter: SChapter) {
+        val comicInfo = parseComicInfoFile(stream)
+
+        comicInfo.title?.let { chapter.name = it.value }
+        comicInfo.number?.value?.toFloatOrNull()?.let { chapter.chapter_number = it }
+        comicInfo.translator?.let { chapter.scanlator = it.value }
     }
 
     // Chapters
     override suspend fun getChapterList(manga: SManga): List<SChapter> = withIOContext {
+        val mangaDir = fileSystem.getMangaDirectory(manga.url) ?: error("${manga.url} is not a valid directory")
+        val mangaDirFiles = mangaDir.listFiles().orEmpty()
+        val noXmlFile = mangaDirFiles
+            .firstOrNull { it.name == ".noxml" }
```

Comment:
```suggestion
        val mangaDir = fileSystem.getMangaDirectory(manga.url) ?: error("${manga.url} is not a valid directory")
        val noXmlExists = mangaDir.listFiles()?.any { it.name == ".noxml" } ?: false 
```

Then `else if (!noXmlExists)`